src/components/translation-builder/translation-builder.component.tsx

```diff
@@ -109,6 +104,39 @@ const TranslationBuilder: React.FC<TranslationBuilderProps> = ({ formSchema, onU
     },
     [fallbackStrings],
   );
+  useEffect(() => {
+    if (selectedLanguageCode === 'en' && formSchema) {
+      setTranslations(fallbackStrings);
+    }
+  }, [selectedLanguageCode, formSchema, fallbackStrings]);
+
+  const languageChanger = async (newLangCode: string) => {
+    setSelectedLanguageCode(newLangCode);
```

Comment:
Shouldn't the loading be set to true at the very start?
```suggestion
    setLoading(true);
    setSelectedLanguageCode(newLangCode);
```

Reply:
No, @NethmiRodrigo, we shouldn't do that because if the selected language is English, the loading state will remain true indefinitely, causing an infinite loading issue.